core/shared/src/main/scala/zio/prelude/Equal.scala

```diff
@@ -862,8 +862,8 @@ object Equal extends EqualVersionSpecific {
    * `Hash` and `Ord` (and thus also `Equal`) instance for `Unit` values.
    * Since there is only one `Unit` value all equality comparisons will always be true.
    */
-  implicit val UnitHashOrd: Hash[Unit] with Ord[Unit] =
-    HashOrd.make(_.hashCode, (_, _) => Ordering.Equals, (_, _) => true)
+  implicit lazy val UnitHashOrd: Hash[Unit] with Ord[Unit] =
+    HashOrd.make(Hash.default.hash, (_, _) => Ordering.Equals, (_, _) => true)
```

Comment:
`().hashCode` generates a warning: "dubious usage of method hashCode with unit value"
I'm not sure what it means.
According to https://www.scala-lang.org/api/2.13.11/scala/Unit.html,
> There is only one value of type Unit, (), and it is not represented by any object in the underlying runtime system
Which would mean that calling `.hasCode` on `()` doesn't have much sens as there's not instance of `Unit` at runtime

I'm not sure why this code was compiling and isn't anymore
I'm also not sure about my fix. I know it removes the warning but that's all I know for now.

Not that I made this instance `lazy` because `Hash.default` is itself `lazy`
I'm sure how much this instance is used TBH